File: src/main/java/ai/aimachineserver/utils/SecurityUtils.java
package ai.aimachineserver.utils;

import ai.aimachineserver.domain.user.User;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

@Component
public class SecurityUtils {
    public Long getLoggedUserId() {
        User loggedUser = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        return loggedUser.id;
    }
}

File: src/main/kotlin/ai/aimachineserver/utils/SecurityUtils.kt
package ai.aimachineserver.utils

import ai.aimachineserver.domain.user.User
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class SecurityUtils {
    fun getLoggedUserId(): Long? {
        val loggedUser = SecurityContextHolder.getContext().authentication.principal as User
        return loggedUser.id
    }
}